### DISCUSSION TITLE
RuntimeError: Concurrent call to receive() is not allowed

### DISCUSSION THREAD

**Discussion by giorno420** [Rapptz/discord.py#6075] [Q&A] [answered]
when i made a discord bot for voice chat, it raised this:

    'Concurrent call to receive() is not allowed')
RuntimeError: Concurrent call to receive() is not allowed

the code is here below:

```
import discord
import youtube_dl
from discord.ext import commands

players = {}
token = nope
client = commands.Bot(command_prefix='%')


@client.event
async def on_ready():
    print('bot ready')


@client.command(pass_context=True)
async def join(ctx):
    channel = ctx.message.author.voice.channel
    await client.connect()


@client.command(pass_context=True)
async def leave(ctx):
    guild = ctx.message.guild
    voice_client = guild.voice_client
    await voice_client.disconnect()


@client.command(pass_context=True)
async def play(ctx, url):
    guild = ctx.message.guild
    voice_client = guild.voice_client
    player = await voice_client.create_ytdl_player(url)
    players[guild.id] = player
    player.start()


client.run(token)

```

**Comment by Rapptz** [comment, marked as answer]
You are using an old version of the library, 0.16.12, which is unsupported. In order to receive support, it would be a good idea to migrate to the newest version. If you need help migrating, visit the migration guide: https://discordpy.readthedocs.io/en/latest/migrating.html

If you need more help consider joining the [official Discord server](https://discord.gg/r3sSKJJ). Please do not open more issues regarding migration or the old version. Thank you.

Also, please do not show your token in public. It would be wise to reset it.

**Reply by giorno420** [reply]
i realised about the token now sorry and thanks for resolving the issue
